internal/pkg/cli/pipeline_delete.go

```diff
@@ -130,65 +136,61 @@ func (o *deletePipelineOpts) Execute() error {
 	return nil
 }
 
-func (o *deletePipelineOpts) readPipelineManifest() error {
-	data, err := o.ws.ReadPipelineManifest()
+func (o *deletePipelineOpts) getNameAndSecret() error {
+	path, err := o.ws.PipelineManifestLegacyPath()
 	if err != nil {
-		if err == workspace.ErrNoPipelineInWorkspace {
-			return err
-		}
-		return fmt.Errorf("read pipeline manifest: %w", err)
+		return err
 	}
-
-	pipeline, err := manifest.UnmarshalPipeline(data)
+	manifest, err := o.ws.ReadPipelineManifest(path)
 	if err != nil {
-		return fmt.Errorf("unmarshal pipeline manifest: %w", err)
+		return err
 	}
-
-	o.PipelineName = pipeline.Name
-
-	if secret, ok := (pipeline.Source.Properties["access_token_secret"]).(string); ok {
-		o.PipelineSecret = secret
+	if o.name == "" {
```

Comment:
This doesn't look right because I think we shouldn't override the value they provided by flag...so if `o.name` was set, which means it was set by the flag, and we should validate in `Validate()` func. And in `Ask()` I think we should assume `o.name` is correct and use it as the pipeline name to get the pipeline manifest.

Reply:
🤔  I think I know what you're getting at, but we have to read the manifest either way, so this way we don't read it twice.

Reply:
Oh, looks like you're caching the manifest in #3283, so that changes things here. Will adjust.